src/server/coprocessor/executor/scanner.rs

```diff
@@ -48,13 +48,13 @@ impl<'a> Scanner<'a> {
     }
 
     pub fn next_row(&mut self, range: &KeyRange) -> Result<Option<(Vec<u8>, Value)>> {
+        if range.get_start() > range.get_end() {
```

Comment:
why move here? I think it's both ok  to do this after or before checking `seek_key`. And In the next refactor, we may remove this `range check` just like `tidb` do.  Since your PR is complicated enough, I prefer to avoid any unnecessary refactor?